src/io.jl

```diff
@@ -74,6 +75,75 @@ hg_save(
     open(io -> hg_save(io, h, format), fname, "w")
 
 
+"""
+    hg_save(io::IO, h::Hypergraph, format::HIF_Format)
+
+Saves a hypergraph `h` to an output stream `io` in `HIF` format.
+
+If `h` has `Composite Types` either for vertex metadata or hyperedges metadata,
+the user has to explicit tell the JSON3 package about it, for instance using:
+
+`JSON3.StructType(::Type{MyType}) = JSON3.Struct()`.
+
+See the (JSON3.jl documentation)[https://github.com/quinnj/JSON3.jl] for more details.
+
+"""
+function hg_save(io::IO, h::Hypergraph, format::HIF_Format)
+    _ = format
+
+    json_hg = Dict{Symbol, Any}()
+    incidences = []
+    v_meta = h.v_meta
+    he_meta = h.he_meta
+
+    if any(isnothing, h.v_meta)
+        v_meta = [i for i in 1:length(h.v_meta)]
```

Comment:
what is the intention of this code?
SimpleHypegraphs supports both `Nothing` as type of metadata (in that case there is no metadata) as well as concrete types

Reply:
Yes, but as far as I understand, we cannot have `Nothing` in edge or node metadata in the HIF format. This is probably not the best take, but the idea was that if there are missing metadata points, then we need to fallback to the solution, where edge or node ids are indexes in the matrix